source/ShipJumpNavigation.cpp

```diff
@@ -71,17 +71,25 @@ void ShipJumpNavigation::SetSystem(const System *system)
 
 
 // Get the amount of fuel that would be expended to jump to the destination. If the destination is
-// nullptr then return the maximum amount of fuel that this ship could expend in one jump.
+// nullptr then return the minimum amount of fuel that this ship could expend in one jump from the
+// current system.
 double ShipJumpNavigation::JumpFuel(const System *destination) const
 {
-	// A currently-carried ship requires no fuel to jump, because it cannot jump.
-	if(!currentSystem)
+	// A currently-carried ship or ship without drives requires no fuel to jump,
+	// because it cannot jump.
+	if(currentSystem != nullptr && !(hasJumpDrive || hasHyperdrive || hasScramDrive))
 		return 0.;
 
-	// If no destination is given, return the maximum fuel per jump.
-	if(!destination)
-		return max(JumpDriveFuel(), HyperdriveFuel());
-
+	// If no destination is given, return the minimum jump-fuel to the nearest system.
+	// If no system is accessible via hyperdrive, return assuming jump drive usage.
+	if(destination == nullptr)
+	{
+		if(!currentSystem->Links().empty() && (hasHyperdrive || hasScramDrive))
+		{
+			return HyperdriveFuel();
+		}
+		return JumpDriveFuel();
+	}
```

Comment:
It is possible for a jump drive outfit to consume less fuel than a hyperdrive or scram drive, so this doesn't necessarily give the minimum.
I'm not really sure about changing this behaviour in general, though. This method is used in a lot of places; are you sure this change doesn't have an undesirable effect on behaviour everywhere else?

Reply:
Fair enough, I've now made a new specific function for this specific mechanism instead.